src/storage/cloud_storage/mod.rs

```diff
@@ -290,40 +290,39 @@ impl<U: Sync + Send> StorageBackend<U> for CloudStorage {
         Box::new(result)
     }
 
-    fn put<P: AsRef<Path>, B: tokio::prelude::AsyncRead + Send + 'static>(
+    async fn put<P: AsRef<Path> + Send, B: tokio::prelude::AsyncRead + Send + 'static>(
         &self,
         _user: &Option<U>,
         bytes: B,
         path: P,
         _start_pos: u64,
-    ) -> Box<dyn Future<Item = u64, Error = Error> + Send> {
-        let uri = match self.uris.put(path) {
-            Ok(uri) => uri,
-            Err(err) => return Box::new(future::err(err)),
-        };
+    ) -> super::Result<u64> {
+        use futures03::compat::*;
```

Comment:
this one is already at the top of the file, but we need a cleanup anyway after everything i s converted, so it doesn't really matter

Reply:
Removed